ARC MWDT toolchain support is broken

Every attempt to build Zephyr v4.1.0-rc1 with ARC MWDT toolchain ends up with failure while building one or another assembly file, like:
```
In file included from /home/tagunil/tests2/zephyr/arch/arc/core/fast_irq.S:18:
In file included from /home/tagunil/tests2/zephyr/include/zephyr/toolchain.h:44:
In file included from /home/tagunil/tests2/zephyr/include/zephyr/toolchain/mwdt.h:18:
/home/tagunil/tests2/zephyr/include/zephyr/toolchain/common.h:101:6: error: Architecture unsupported
  101 |     #error Architecture unsupported
```
I've bisected it down to 65471a6095fd46d537dd4b7ad0667b4f1018a1c2. I'm not sure why our toolchain is the only one affected, need to look into it further.

--- a/cmake/compiler/arcmwdt/compiler_flags.cmake
+++ b/cmake/compiler/arcmwdt/compiler_flags.cmake
@@ -165,6 +165,9 @@ set_compiler_property(PROPERTY coverage "")
 # mwdt compiler flags for imacros. The specific header must be appended by user.
 set_compiler_property(PROPERTY imacros -imacros)
 
+# assembler compiler flags for imacros. The specific header must be appended by user.
+set_property(TARGET asm PROPERTY imacros -imacros)
+
 # Security canaries.
 #no support of -mstack-protector-guard=global"
 set_compiler_property(PROPERTY security_canaries -fstack-protector)

